protocol/messenger.go

```diff
@@ -2009,8 +2021,8 @@ func (m *Messenger) handleRetrievedMessages(chatWithMessages map[transport.Filte
 		}
 	}
 
-	if len(messageState.Response.Contacts) > 0 {
-		err = m.persistence.SaveContacts(messageState.Response.Contacts)
+	if len(contactsToSave) > 0 {
+		err = m.persistence.SaveContacts(contactsToSave)
```

Comment:
Wouldn't using `messageState.ModifiedContacts` here make sense?

Reply:
`ModifiedContacts` contains only the ID of the one that have  been modified, we need to hydrate them (pull them from `allContacts`)